src/intersection-observer-polyfill.js

```diff
@@ -180,51 +192,65 @@ export class IntersectionObserverPolyfill {
         this.threshold_[this.threshold_.length - 1] <= 1,
         'Threshold should be in the range from "[0, 1]"');
 
-    /** @private {?Element} */
-    this.element_ = null;
-
     /**
-     * The prev threshold slot which the previous ratio fills
-     * Range [0, this.threshold_.length]
-     * TODO: Do we always want to call callback at the first tick?
-     * @private {number}
+     * Store a list of observed elements and their current threshold slot which
+     * their intersection ratio fills, range from [0, this.threshold_.length]
+     * @private {Array<!observeEntryDef>}
      */
-    this.prevThresholdSlot_ = 0;
-
-    // TODO: create the PositionObserver class, or create listener for position
-    // info if inside an iframe
-
-    // TODO: Add the postMessageApi logic outside IntersectionObserverPolyfill.
+    this.observeEntries_ = [];
   }
 
   /**
-   * Provide a way to observer the intersection change for a specific element
-   * Please note that the IntersectionObserverPolyfill only allow to observe one
-   * element at a time.
-   * TODO: Support observing multiple elements.
+   * Provide a way to observe the intersection change for a specific element
+   * Please note IntersectionObserverPolyfill only support AMP element now
    * TODO: Support non AMP element
    * @param {!Element} element
    */
   observe(element) {
     // Check the element is an AMP element
     dev().assert(element.isBuilt && element.isBuilt());
-    // Reset cached value for a new element
 
-    this.prevThresholdSlot_ = 0;
-    this.element_ = element;
+    // If the element already exists in current observeEntries, do nothing
+    this.observeEntries_.forEach(entry => {
+      if (entry.element === element) {
+        dev().warn('INTERSECTION-OBSERVER', 'should observe same element once');
+      }
+      return;
```

Comment:
No, native one won't generate two changes for exact same elements. When user observe the same element for the second time, it will simply ignore the second `#observe` call.